cg/meta/workflow/balsamic.py

```diff
@@ -205,23 +227,44 @@ def get_verified_bed(self, sample_data: dict, panel_bed: Path) -> Optional[str]:
                 )
             return Path(self.bed_path, target_bed).as_posix()
 
-    @staticmethod
-    def get_verified_pon(panel_bed: Path, pon_cnn: str) -> Optional[str]:
-        """Returns the validated PoN path
+    def get_verified_pon(self, panel_bed: str, pon_cnn: str) -> Optional[str]:
+        """Returns the validated PON or extracts the latest one available if it is not provided
 
         Raises BalsamicStartError:
-            When there is a missmatch between the PoN and the panel bed file names
+            When there is a missmatch between the PON and the panel bed file names
         """
-        if pon_cnn:
-            pon_cnn = Path(str(pon_cnn))
-            if panel_bed.stem not in pon_cnn.stem:
+
+        if not panel_bed:
+            latest_pon = None
+        elif pon_cnn:
+            latest_pon = pon_cnn
+            if Path(panel_bed).stem not in Path(latest_pon).stem:
                 raise BalsamicStartError(
-                    f"The specified PoN reference file {pon_cnn} does not match the panel bed {panel_bed}"
+                    f"The specified PON reference file {latest_pon} does not match the panel bed {panel_bed}"
+                )
+        else:
+            latest_pon = self.get_latest_pon_file(panel_bed)
+            if latest_pon:
+                LOG.warning(
+                    f"The following PON reference file will be used for the analysis: {latest_pon}"
                 )
 
-            return pon_cnn.as_posix()
+        return latest_pon
+
+    def get_latest_pon_file(self, panel_bed: str) -> Optional[str]:
+        """Returns the latest PON cnn file associated to a specific capture bed"""
 
-        return pon_cnn
+        if not panel_bed:
+            raise BalsamicStartError("BALSAMIC PON workflow requires a panel bed to be specified")
+
+        pon_list = Path(self.bed_pon_path).glob(
+            f"*{Path(panel_bed).stem}_CNVkit_PON_reference_v*.cnn"
+        )
+        sorted_pon_files = sorted(
+            pon_list, key=lambda file: int(file.stem.split("_v")[-1]), reverse=True
```

Comment:
Not incredibly important but you could use these instead of the numbers:
https://github.com/Clinical-Genomics/cg/blob/master/cg/constants/indexes.py

Reply:
Added ^^